cpp/oneapi/dal/backend/primitives/sort/sort_dpc.cpp

```diff
@@ -0,0 +1,529 @@
+                                                  part_hist_,
+                                                  part_prefix_hist_,
+                                                  local_size_,
+                                                  local_hist_count_,
+                                                  scan_deps);
+            res_deps = radix_reorder(queue_,
+                                     val_in,
+                                     ind_in,
+                                     part_prefix_hist_,
+                                     val_buff_,
+                                     ind_buff_,
+                                     elem_count_,
+                                     bit_offset,
+                                     local_size_,
+                                     local_hist_count_,
+                                     hist_scan_deps);
+        }
+        else {
+            auto scan_deps = radix_scan(queue_,
+                                        val_buff_,
+                                        part_hist_,
+                                        elem_count_,
+                                        bit_offset,
+                                        local_size_,
+                                        local_hist_count_,
+                                        res_deps);
+            auto hist_scan_deps = radix_hist_scan(queue_,
+                                                  part_hist_,
+                                                  part_prefix_hist_,
+                                                  local_size_,
+                                                  local_hist_count_,
+                                                  scan_deps);
+            res_deps = radix_reorder(queue_,
+                                     val_buff_,
+                                     ind_buff_,
+                                     part_prefix_hist_,
+                                     val_in,
+                                     ind_in,
+                                     elem_count_,
+                                     bit_offset,
+                                     local_size_,
+                                     local_hist_count_,
+                                     hist_scan_deps);
+        }
+    }
+
+    sort_event_ = res_deps;
+    return res_deps;
+}
+
+template <typename Integer>
+radix_sort<Integer>::radix_sort(const sycl::queue& queue, std::int64_t vector_count)
+        : queue_(queue),
+          vector_count_(0) {
+    init(queue_, vector_count);
+}
+
+template <typename Integer>
+radix_sort<Integer>::~radix_sort() {
+    sort_event_.wait_and_throw();
+}
+
+template <typename Integer>
+void radix_sort<Integer>::init(sycl::queue& queue, std::int64_t vector_count) {
+    std::uint32_t uint_vector_count = de::integral_cast<std::uint32_t>(vector_count);
+    if (uint_vector_count && vector_count_ != uint_vector_count) {
```

Comment:
I assume `uint_vector_count` is always > 0, so just put assertion.

Reply:
fixed